docs: updating cli upgrade versions and linting (#4137)
mesosphere/dcos-docs-site: pages/dkp/konvoy/1.8/upgrade/upgrade-cli/index.md

After you have the available Konvoy versions, you can upgrade your CLI by running the following command:

```bash
konvoy image upgrade --version=v1.8.5
Wrote Konvoy CLI version 'v1.8.5' to '.konvoy/cli_version'
```

After the upgrade command completes, you can start using the new Konvoy version.

### Upgrading Konvoy from v1.7.x to v1.8.5

**You must modify your `cluster.yaml` with these changes when upgrading from a previous Konvoy version:**

> It is recommended to upgrade to the newest supported version of Kubernetes. Set `spec.kubernetes.version: 1.20.13` for of kind: ClusterConfiguration.

Reviewer: I've never noticed this before, but the phrasing 'for of kind' in all these items is very strange and awkward.

```suggestion
It is recommended to upgrade to the newest supported version of Kubernetes. Set `spec.kubernetes.version: 1.20.13` for kind: ClusterConfiguration.
```

> The version of Calico got bumped in Konvoy v1.8.5, from Calico v3.17.3 to v3.20.2. You will want to change this in `spec.containerNetworking.calico.version: v3.17.3` for of kind: ClusterConfiguration.

Reviewer: I believe this is incorrect, the Calico bump happened in 1.8.4, not 1.8.5.  

```suggestion
The version of Calico was bumped in Konvoy v1.8.4, from Calico v3.17.3 to v3.20.2.  When upgrading, set `spec.containerNetworking.calico.version: v3.20.2` for kind: ClusterConfiguration.
```

Author: Good catch with that wording @cneth

> The version of Kubernetes Base Addons changed if you use KBA, so you need to change your `configVersion` for your `configRepository`: `https://github.com/mesosphere/kubernetes-base-addons` to be `spec.addons.configVersion: stable-1.20-4.4.0` for of kind: ClusterConfiguration.

Reviewer: ```suggestion
The version of Kubernetes Base Addons changed if you use KBA, so you need to change your `configVersion` for your `configRepository`: `https://github.com/mesosphere/kubernetes-base-addons` to be `spec.addons.configVersion: stable-1.20-4.4.0` for kind: ClusterConfiguration.
```

> If you use Kommander, you need to change the `configVersion` for your `configRepository`: `https://github.com/mesosphere/kubeaddons-kommander` to be `spec.addons.configVersion: stable-1.20-1.4.4` for of kind: ClusterConfiguration.

Reviewer: ```suggestion
If you use Kommander, you need to change the `configVersion` for your `configRepository`: `https://github.com/mesosphere/kubeaddons-kommander` to be `spec.addons.configVersion: stable-1.20-1.4.4` for kind: ClusterConfiguration.
```

> If you have Dispatch enabled, you need to change the `configVersion` for your `configRepository`: `https://github.com/mesosphere/kubeaddons-dispatch` to be `configVersion: stable-1.20-1.4.6` for of kind: ClusterConfiguration

Reviewer: ```suggestion
If you have Dispatch enabled, you need to change the `configVersion` for your `configRepository`: `https://github.com/mesosphere/kubeaddons-dispatch` to be `configVersion: stable-1.20-1.4.6` for kind: ClusterConfiguration
```

> The version of Konvoy is now `v1.8.5`, set `spec.version: v1.8.5` for both of kind: ClusterProvisioner and of kind: ClusterConfiguration.

Reviewer: ```suggestion
The version of Konvoy is now `v1.8.5`, set `spec.version: v1.8.5` for both kind: ClusterProvisioner and kind: ClusterConfiguration.
```
